ambari-server/src/main/java/org/apache/ambari/server/controller/ControllerModule.java

```diff
@@ -640,25 +640,27 @@ private void installFactories() {
     // the dispatch factory
     for (BeanDefinition beanDefinition : beanDefinitions) {
       String className = beanDefinition.getBeanClassName();
-      Class<?> clazz = ClassUtils.resolveClassName(className,
-          ClassUtils.getDefaultClassLoader());
-
-      try {
-        NotificationDispatcher dispatcher;
-        if (clazz.equals(AmbariSNMPDispatcher.class)) {
-          dispatcher = (NotificationDispatcher) clazz.getConstructor(Integer.class).newInstance(configuration.getAmbariSNMPUdpBindPort());
-        } else if (clazz.equals(SNMPDispatcher.class)) {
-          dispatcher = (NotificationDispatcher) clazz.getConstructor(Integer.class).newInstance(configuration.getSNMPUdpBindPort());
-        } else {
-          dispatcher = (NotificationDispatcher) clazz.newInstance();
+      if (className != null) {
```

Comment:
Can we add some log message, in case we got  className with null value?

Reply:
Thanks @hapylestat for the review. Updated the patch based on the comment, please let me know if this looks good to you

Reply:
Updated trunk PR: #3037 as well

Reply:
@hapylestat @dlysnichenko could you please help me trigger a build for this PR?

Reply:
@hapylestat @dlysnichenko Could you please review this patch

Reply:
Please review @hapylestat @dlysnichenko @jonathan-hurley

Reply:
Please review @dlysnichenko @hapylestat 
PR for trunk: #3037

Reply:
Please review @dlysnichenko @hapylestat
PR for trunk: #3037

Reply:
Please review @mpapirkovskyy